### DISCUSSION TITLE
Unable to train classifier with padding=1 using PyTorch

### DISCUSSION THREAD

**Discussion by KarthikGanesan88** [Trusted-AI/adversarial-robustness-toolbox#1346] [Q&A] [answered]
**Describe the bug**
When I try to train a classifier with a padding value = 1, I get an error:
"ValueError: Expected input batch_size (100) to match target batch_size (64)." I am able to train a classifier using the default value of 0 padding. 

**To Reproduce**
1. Download the '[get_started_pytorch.py](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/examples/get_started_pytorch.py)' file. 
2. Edit lines 23 and 24 to add 'padding=1' as an extra paramter.
3. Run the script. 

**Expected behavior**
Successfully trained network. 

**System information:**
 - OS : Ubuntu 20.04 LTS
 - Python version: 3.9.6
 - ART version : 1.8.0
 - PyTorch version: 1.9.0

**Comment by beat-buesser** [comment, marked as answer]
Hi @KarthikGanesan88 Thank you very much for using ART!

I don't think that this error is caused by ART.

It looks like it is caused by your modification to the convolution layers and you will have to update the layers after the convolution layers accordingly.

**Reply by KarthikGanesan88** [reply]
Oh, right. I understand. I didn't modify the x.view layer after the conv. Thank you!
